app/routes/customers_routes.py

```diff
@@ -0,0 +1,135 @@
+from flask import Blueprint
+from app import db
+from app.models.customer import Customer
+from app.models.rental import Rental
+from flask import Blueprint, jsonify, make_response, request, abort
+import datetime, requests, os
+import os
+
+customers_bp = Blueprint("customers_bp", __name__, url_prefix="/customers")
+
+# def valid_int(number, parameter_type):
+#     try:
+#         int(number)
+#     except:
+#        return {"message": "Invalid data"}, 400
+
+# def get_customer_from_id(customer_id):
+#     valid_int(customer_id, "customer_id")
+#    #return Customer.query.get_or_404({"message":f"Customer {customer_id} was not found"}, 404 
+#     return Customer.query.get_or_404(customer_id, description="{customer not found}")
+
+
+# @customers_bp.route("/<customer_id>", methods=["GET"])
+# def get_customer_by_id(customer_id):
+#     single_customer = get_customer_from_id(customer_id)
+#     return {single_customer.to_dict()}, 200
+
+@customers_bp.route("/<id>", methods=["GET"])
+def read_one_customer(id):
+    try:
+        int(id)
+    except:
+        return {"message": "Invalid data"}, 400
+
+    customer = Customer.query.get(id)
+
+    if not customer:
+        return {"message": f"Customer {id} was not found"}, 404        
+
+    return customer.to_dict(), 200
+
+@customers_bp.route("/<id>/rentals", methods=["GET"])
+def read_one_customer_rentals(id):
+    try:
+        int(id)
+    except:
+        return {"message": "Invalid data"}, 400
+
+    customer = Customer.query.get(id)
+
+    if not customer:
+        return {"message": f"Customer {id} was not found"}, 404        
+
+    rentals = customer.rentals
+    response_body  = [rental.get_video() for rental in rentals]
+    return make_response(jsonify(response_body), 200)
+
+
+@customers_bp.route("", methods=["GET"])
+def get_all_cusotmers():
+
+    customers = Customer.query.all()
+
+    response_body = []
+
+    if not customers:
+        return jsonify([]), 200
+
+    for customer in customers:
+        response_body.append(customer.to_dict())
+    
+    return jsonify(response_body), 200
+
+
+@customers_bp.route("", methods=["POST"])
+def create_customer():
+    request_body = request.get_json()
+
+    missing_data = ""
+    if "name" not in request_body:
+        missing_data = "name"
+    elif "postal_code" not in request_body:
+        missing_data = "postal_code"
+    elif "phone" not in request_body:
+        missing_data = "phone"
```

Comment:
Minor note. Any or all of this data might be missing. Consider using if/if/if and collecting the missing data in a list, then joining in the return.